core/store/src/trie/mod.rs

```diff
@@ -612,16 +606,11 @@ impl Trie {
         }
     }
 
-    pub(crate) fn retrieve_raw_bytes(&self, hash: &CryptoHash) -> Result<Arc<[u8]>, StorageError> {
```

Comment:
@matklad - more Rust question:

in such case - should we 'keep' this method (and have it simply call self.storage.retrieve_raw_bytes) or should we allow other classes to call XX.storage.retrieve_raw_bytes (as Alex did in this PR)?

For me - one advantage of keeping this method - is that we keep better 'layers' - and avoid having other classes 'peek' into Trie) - but I wonder if it has some performance implications

Reply:
There won't be performance implications here. In general, it's a pretty safe assumption in Rust that linguistic abstractions don't introduce overhead -- static nature of the language and unboxed memory layout guarantee that compiler "sees through" any abstractions. 

Regarding layering, my general observation is that layering is good only when it is enforced: it's better to focus on specific places where layering is important, rather than to try to enforce it recursively throughout the stack. There should be places where there's explicitly *no* layers.

For Rust, the best enforcement is aligning layers with crate boundaries: that's why I am relatively picky about `pub` vs `pub(crate)`, but relatively indifferent between `pub(crate)` and `/*private*/`.

As an aside, I think we can make storage private rather `pub(crate)` if we put trie tests into submodule of the `trie` module. We only need `pub(crate)` for tests, but "`pub(crate)` for tests" is a bit of oxymoron -- `pub(crate)` guarantees that this is for *unit* tests, and unit tests can always be made a submodule (in Rust, submodule see the full contents of partent modules)